audit/archive/reader.go

```diff
@@ -275,37 +286,26 @@ func NewReader(archivePath string) (*Reader, error) {
 
 	// Returns a deduplicated list of tags for the specific label present in the archive
 	// e.g. getUniqueTags(serverTagLabel) -> [Tag(server, s1), Tag(server, s2, Tag(server, s3)]
-	// TODO each call scans the manifest.
-	// Ok for now, but a single scan could build a list of unique tag values for each tag name
 	getUniqueTags := func(label TagLabel) []Tag {
-		var tagsSet = make(map[Tag]struct{}, len(manifestMap))
-		for _, tags := range manifestMap {
-			for _, tag := range tags {
-				if tag.Name == label {
-					// Found a tag for the given label, add it to the set
-					tagsSet[tag] = struct{}{}
-				}
+		var tags []Tag
+		for tag := range invertedIndex {
+			if tag.Name == label {
+				tags = append(tags, tag)
 			}
 		}
-		// Create list of unique tags from the set
-		tagsList := make([]Tag, 0, len(tagsSet))
-		for tag := range tagsSet {
-			tagsList = append(tagsList, tag)
-		}
-		slices.SortFunc(tagsList, func(a, b Tag) int {
+		slices.SortFunc(tags, func(a, b Tag) int {
 			if a.Name != b.Name {
 				panic("Unexpected comparison between different tags")
```

Comment:
This isnt new things but can you please get rid of this panic here somehow? Previously this was not usable as a library but since it is now we should avoid panics.